src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -148,13 +148,36 @@ class SignUpModalController {
             }
             onSuccess(postData)
           }
+        },
+        // TODO: make sure that this callback is only called after the sign up completes successfully
+        postSubmit: (postData, onSuccess) => {
+          onSuccess(postData)
+
+          const donorDetails = {
+            name: {
+              'given-name': postData.userProfile.firstName,
+              'family-name': postData.userProfile.lastName
+            },
+            email: postData.userProfile.email,
+            phone: postData.userProfile.primaryPhone,
+            mailingAddress: {
+              streetAddress: postData.userProfile.streetAddress,
+              locality: postData.userProfile.city,
+              region: postData.userProfile.state,
+              postalCode: postData.userProfile.zipCode,
+              country: postData.userProfile.countryCode
+            }
+          }
+          this.$scope.$apply(() => this.onSignUp({ donorDetails }))
         }
       }
     })
 
     this.signIn()
 
     this.oktaSignInWidget.on('afterRender', this.afterRenderFn.bind(this))
+
+    this.$scope.$apply(() => this.onSignUp())
```

Comment:
Could you explain why you need to run `this.onSignUp()` here?

Reply:
Good catch! That was accidentally left in for testing